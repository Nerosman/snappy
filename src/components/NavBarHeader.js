import React from 'react';
import {Collapse, Nav, Navbar, NavbarBrand, NavbarToggler} from 'reactstrap';
import {TitleComponent} from "./TitleComponent";
import {NavItemsComponent} from "./NavItemsComponent";

export class NavBarHeaderComponent extends React.Component {
    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            isOpen: false
        };
    }
    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        const {pathname} = window.location;
        return (
            <div>
                <Navbar color="white" light expand="md">
                    <NavbarBrand href="/"><TitleComponent pathname={pathname}/></NavbarBrand>
                    <NavbarToggler onClick={this.toggle} />
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItemsComponent pathname={pathname}/>
                        </Nav>
                    </Collapse>
                </Navbar>
            </div>
        );
    }
}