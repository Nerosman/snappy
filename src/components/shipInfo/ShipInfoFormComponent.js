import React from "react";
import {InputComponent} from "../reusableComponents/InputComponent";
import {Button, Form} from "reactstrap";
import {validateAddress, validateEmail, validateLengthOfValue} from "../../utils/validations";
import {toast} from "react-toastify";
import {addOrderRequest} from "../../communicator";

export class ShipInfoFormComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailValidity: true,
            phoneValidity: true,
            specialNotesValidity: true,
            addressValidity: true,
            address1Validity: true,
            address2Validity: true,
            firstNameValidity: true,
            secondNameValidity: true,
            address1: "",
            address2: "",
            firstName: "",
            secondName: "",
            phone: "",
            specialNotes: "",
            email: ""

        }
    }

    inputValueToState(e) {
        this.setState({[e.target.name]: e.target.value})
    }

    emailParameterToState(e) {
        const email = e.target.value;
        const emailValidity = validateEmail(email);
        this.setState({
            [e.target.name]: email,
            emailValidity
        });
    }

    phoneParameterToState(e) {
        const phone = e.target.value;
        const phoneValidity = validateLengthOfValue(phone, 10);
        this.setState({
            [e.target.name]: phone,
            phoneValidity
        });
    }

    specialNotesToState(e) {
        const specialNotes = e.target.value;
        const specialNotesValidity = validateLengthOfValue(specialNotes, 500);
        this.setState({
            [e.target.name]: specialNotes,
            specialNotesValidity
        });
    }

    submitForm(e) {
        e.preventDefault();
        // validateAddress(this.state.address1, this.state.address2)
        //     .then(response => response.json())
        //     .then(jsondata => {
        //         this.setState({
        //             addressValidity: jsondata.ErrorCode === 0,
        //             addressErrorMessage: jsondata.ErrorMessage
        //         })
        //     })
        let inputsValidity = [];

        for (let key in this.state) {
            if (key.includes("Validity")) {
                inputsValidity.push(this.state[key])
            }
        }

        const isFormNonValid = inputsValidity.includes(false);

        if (isFormNonValid) {
            toast("Something went wrong. Please check your data", {
                position: toast.POSITION.BOTTOM_RIGHT,
                type: "error"
            })
        } else {
            addOrderRequest(this.state).then((res) => {
                res.status === 200 && toast("Your date has been save successfully", {
                    position: toast.POSITION.BOTTOM_RIGHT,
                    type: "success"
                })
            })
        }
    }

    render() {
        return <Form onSubmit={this.submitForm.bind(this)}>
            <InputComponent
                id="first-name"
                name="firstName"
                label="First name"
                type="text"
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.firstNameValidity}
                placeholder="John"
                errorMessage={"Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="second-name"
                name="secondName"
                label="Second name"
                type={"text"}
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.secondNameValidity}
                placeholder="Whiskey"
                errorMessage={"Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="address"
                name="address1"
                label="Address"
                type={"text"}
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.addressValidity && this.state.address1Validity}
                placeholder="1234 Main St"
                errorMessage={this.state.addressErrorMessage || "Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="cityStateZip"
                name="address2"
                label="City, State, Zip"
                type="text"
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.addressValidity && this.state.address2Validity}
                placeholder="ASBURY PARK, NJ 07712-6086 "
                errorMessage={this.state.addressErrorMessage || "Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="email"
                name="email"
                label="Email"
                type={"email"}
                onChange={this.emailParameterToState.bind(this)}
                onValid={this.state.emailValidity}
                errorMessage="Please include @ and . in the email address"
                required={true}
            />
            <InputComponent
                id="phone"
                name="phone"
                label="Phone"
                onChange={this.phoneParameterToState.bind(this)}
                onValid={this.state.phoneValidity}
                errorMessage="Length of phone number must be less then 10 charts"
                required={true}
            />
            <InputComponent
                id="special-notes"
                name="specialNotes"
                label="Special notes"
                type={"textarea"}
                onChange={this.specialNotesToState.bind(this)}
                onValid={this.state.specialNotesValidity}
                errorMessage="Length of special notes must be less then 500  charts"
            />
            <Button id="submit-ship-info-button" type="submit">Submit</Button>
        </Form>
    }
}
