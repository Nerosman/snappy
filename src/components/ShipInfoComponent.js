import React from "react";
import {Container} from "reactstrap";
import {ShipInfoFormComponent} from "./ShipInfoFormComponent";

export const ShipInfoComponent = () => {
    return <Container>
        <ShipInfoFormComponent/>
    </Container>
}