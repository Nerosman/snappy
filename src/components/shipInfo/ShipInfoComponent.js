import React from "react";
import {Col, Container, Row} from "reactstrap";
import {ShipInfoFormComponent} from "./ShipInfoFormComponent";

export const ShipInfoComponent = () => {
    return <Container fluid>
        <Row lg={12}>
            <Col lg={{ size: 6, offset: 3 }} className="well">
                <legend>
                    Fill shipping info to receive a gift
                </legend>
                <ShipInfoFormComponent/>
            </Col>
        </Row>
    </Container>
}