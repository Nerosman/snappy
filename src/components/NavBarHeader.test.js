import {configure, shallow} from "enzyme/build";
import {ShipInfoComponent} from "./shipInfo/ShipInfoComponent";
import jest from "jest-mock";
import React from "react";
import Adapter from "enzyme-adapter-react-16/build";
import {NavBarHeaderComponent} from "./NavBarHeader";

configure({ adapter: new Adapter() });

describe('<NavBarHeaderComponent />', () => {
    test('take snapshot for header', () => {
        const wrapper = shallow(
            <NavBarHeaderComponent />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('could save input value to state', ()=> {
        const wrapper = shallow(<NavBarHeaderComponent />);
        const instance = wrapper.instance();
        jest.spyOn(instance, 'toggle');
        wrapper.find('NavbarToggler').simulate('click');
        expect(wrapper.state('isOpen')).toEqual(true);
    });
});
