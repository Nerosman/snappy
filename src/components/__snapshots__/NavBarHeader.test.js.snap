// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarHeaderComponent /> take snapshot for header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBarHeaderComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        color="white"
        expand="md"
        light={true}
        tag="nav"
      >
        <NavbarBrand
          href="/"
          tag="a"
        >
          <TitleComponent
            pathname="/"
          />
        </NavbarBrand>
        <NavbarToggler
          onClick={[Function]}
          tag="button"
          type="button"
        />
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
        >
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItemsComponent
              pathname="/"
            />
          </Nav>
        </Collapse>
      </Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavbarBrand
            href="/"
            tag="a"
          >
            <TitleComponent
              pathname="/"
            />
          </NavbarBrand>,
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItemsComponent
                pathname="/"
              />
            </Nav>
          </Collapse>,
        ],
        "color": "white",
        "expand": "md",
        "light": true,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <TitleComponent
              pathname="/"
            />,
            "href": "/",
            "tag": "a",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "pathname": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItemsComponent
                pathname="/"
              />
            </Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <NavItemsComponent
                pathname="/"
              />,
              "className": "ml-auto",
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "pathname": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          color="white"
          expand="md"
          light={true}
          tag="nav"
        >
          <NavbarBrand
            href="/"
            tag="a"
          >
            <TitleComponent
              pathname="/"
            />
          </NavbarBrand>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItemsComponent
                pathname="/"
              />
            </Nav>
          </Collapse>
        </Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavbarBrand
              href="/"
              tag="a"
            >
              <TitleComponent
                pathname="/"
              />
            </NavbarBrand>,
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            />,
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                className="ml-auto"
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItemsComponent
                  pathname="/"
                />
              </Nav>
            </Collapse>,
          ],
          "color": "white",
          "expand": "md",
          "light": true,
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <TitleComponent
                pathname="/"
              />,
              "href": "/",
              "tag": "a",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "pathname": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "tag": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": <Nav
                className="ml-auto"
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItemsComponent
                  pathname="/"
                />
              </Nav>,
              "enter": true,
              "exit": true,
              "in": false,
              "isOpen": false,
              "mountOnEnter": false,
              "navbar": true,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 350,
              "unmountOnExit": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavItemsComponent
                  pathname="/"
                />,
                "className": "ml-auto",
                "navbar": true,
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "pathname": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
